packages/relay/src/lib/eth.ts

```diff
@@ -2308,7 +2320,7 @@ export class EthImpl implements Eth {
     }
 
     transactionArray = this.populateSyntheticTransactions(showDetails, logs, transactionArray, requestDetails);
-    transactionArray = showDetails ? _.uniqBy(transactionArray, 'hash') : _.uniq(transactionArray);
+    transactionArray = showDetails ? transactionArray : _.uniq(transactionArray);
```

Comment:
I'm not sure `showDetails` should change the length of the array, it concerns only with returning either transaction hashes or the transaction objects.

Reply:
Hmm, I agree with Luis here. The size should be the same regardless of `showDetails`.
I'm not sure why this logic exists, shouldn't they always be unique transactions based on hash?
Maybe something to follow up on, why should we ever have duplicates when `showDetails` is true?

Reply:
Btw, something that could be informative is that in ethereum `WRONG_NONCE` transactions will be discarded and won't make it out of the mempool and thus it's not possible to have such transactions inside a block.

But for `INVALID_ACCOUNT` transactions, it's kind of different. I'm not sure if we want to discard them from the response of `getBlockByHash`/`getBlockByNumber`, even though they are creating duplicated transaction hashes. What is your opinion?

Reply:
>Btw, something that could be informative is that in ethereum WRONG_NONCE transactions will be discarded and won't make it out of the mempool and thus it's not possible to have such transactions inside a block.

Yes, this is expected. In our case, the Relay also quickly fails `WRONG_NONCE` transactions, thanks to `precheck.nonce()`. Transactions with a `WRONG_NONCE` code are stopped at the precheck stage and don’t reach the consensus node, meaning they’re not included in a block. I assume this may have happened in some blocks because we previously had a nonce buffer in the precheck, which allowed transactions with `nonce + 1` to pass the precheck but then ultimately get flagged as `WRONG_NONCE` by consensus nodes and included in a block. However, the nonce buffer was removed in a recent PR, so this shouldn’t occur again.

Reply:
To sum up, it's safe to use the lodash `uniq` method when the `transactionArray` contains only strings of EVM hashes, but when `showDetails` is `true` and the `transactionArray` contains actual objects, we want to show everything that has not been explicitly filtered by the for-loop above. 

If there are still some transaction objects with duplicated hashes, that means we have missed to filter something and we have a bug in the code, otherwise these bugs will be "hidden" if we keep using the `_.uniqBy(transactionArray, 'hash')` method.